tests/lib/sprintf/src/main.c

```diff
@@ -78,13 +78,26 @@ union raw_double_u {
 };
 #endif
 
+static int WriteFrmtd_vf(FILE *stream, char *format, ...)
+{
+	int ret;
+	va_list args;
+
+	va_start(args, format);
+	ret = vfprintf(stream, format, args);
+	va_end(args);
+
+	return ret;
+}
+
 /**
  *
  * @brief Test sprintf with doubles
  *
  */
 
-void test_sprintf_double(void)
+#ifdef CONFIG_STDOUT_CONSOLE
```

Comment:
nit: So my preference would to move the STDOUT console tests to a different file that's conditionally compiled. But that's by no means a blocker to this PR.